### ISSUE TITLE
skipLibCheck doesn't work

### ISSUE THREAD

**Issue by tjx666** [microsoft/TypeScript#53128]
# Bug Report

### 🔎 Search Terms

skipLibCheck

### 🕗 Version & Regression Information

I test 4.9 and latest version "typescript": "5.1.0-dev.20230307", both can reproduce.

### ⏯ Playground Link

https://github.com/tjx666/wpm-issue

### 💻 Code

```ts
import * as wpm from '@growing-web/wpm';

console.log(wpm);
```

One thing need to specify is that the checked lib point type to a ts file:

![image](https://user-images.githubusercontent.com/41773861/223366316-7c394009-935f-42b2-946d-779f3181778b.png)

### 🙁 Actual behavior

check the lib:

![image](https://user-images.githubusercontent.com/41773861/223365854-10b095ea-cad8-4b5f-9b30-d502f49cc4ba.png)

### 🙂 Expected behavior

don't throw type error

**Comment by MartinJohns** [comment]
[skipLibCheck](https://www.typescriptlang.org/tsconfig#skipLibCheck):

> Skip type checking of declaration files.

But you're not dealing with declaration files, you're dealing with source files. The package you're trying to import should release JavaScript files with related TypeScript declaration files, not TypeScript source files.

**Comment by tjx666** [comment]
@MartinJohns  But that ts file is used to as types

**Comment by MartinJohns** [comment]
It's used as source. You're importing it.

**Comment by tjx666** [comment]
@MartinJohns any way to override the packge types?
